feat(bph_shell): Add initial bph_shell and cli

MrKevinWeiss opened the pull request:
This is the initial commit of the cli and shell for the BPH.

The cli is intended to be used for the startup script.  It does basic things like setting pins and power modes.

The shell is intended for developers who want to play around with testing and also for qualification of the BPH board.  It has many helper functions and everything should be available with autocomplete or though the `help` function provided by the cmd library.

Testing will require a raspberry pi with the BPH setup...  Good luck.

close #14 
close #8
fixes #6

MrKevinWeiss commented:
Also may be useful for @smlng

MrKevinWeiss commented:
I guess it also includes the bootloader pr...

MrKevinWeiss commented:
I think the things that I would like feedback on is the filename of the pinmap json file.  I don't like the versioning issues we can run into.

smlng commented on bph_pal/LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2018 RIOT
```
its 2019 already

smlng commented on bph_pal/bph_pal/__main__.py:
```diff
@@ -0,0 +1,63 @@
+
+
+try:
+    from .bph_shell import BphShell as BPH
```
this whole try except looks ugly, I bet there is a nice way (I look into it)

MrKevinWeiss replied:
I agree, I would love a 
`from .bph_shell or bph_pal.bph_shell or bph_shell import BphShell as BPH` but I couldn't find anything that does that yet.

smlng replied:
look at https://github.com/riot-appstore/PHiLIP/blob/master/IF/philip_pal/philip_pal/__main__.py ??

smlng commented on bph_pal/useful_commands.md:
````diff
@@ -0,0 +1,27 @@
+```
+pip3 uninstall -y bph_pal
+rm -rf dist
+python3 setup.py sdist
````
this should go into the README.md

MrKevinWeiss replied:
This is meant more for development.  Most once this is done people should get it from pip.

smlng replied:
mhm okay

MrKevinWeiss commented:
It was inspired by #8, I got a lot of the ina driver from there.

smlng commented on bph_pal/bph_pal/__main__.py:
```diff
@@ -0,0 +1,63 @@
+        BPH().onecmd("power_source " + str(pargs.power_source))
+
+
+if __name__ == '__main__':
```
not needed, because this file is already name `__main__.py` so its implicit that this matches, just call `main()` as in https://github.com/riot-appstore/PHiLIP/blob/master/IF/philip_pal/philip_pal/__main__.py

MrKevinWeiss replied:
Ahh so you left it out on purpose, good!

smlng commented on bph_pal/bph_pal/bph_shell.py:
```diff
@@ -0,0 +1,475 @@
+            voltage_type: The voltage types...
+                          [bus, shunt]
+        """
+        value = (arg or '').split(' ')
```
I think I already  mentioned that somewhere, when you use `split()` instead of `split(' ')` you can omit the next line and directly use `value` or assign that to `func_args`.

smlng replied:
or did you find a problem with that?

smlng commented:
other than that, please squash as you see fit.

MrKevinWeiss commented:
Should be good now!  Let me know if you want to test together!

MrKevinWeiss commented:
Cleared up the ina226 sensing issues.  Now I truncate negative numbers to 0 since any negative number would just be noise, I fixed the high/low range current range settings (changing the bypass resistor) and fixed the shunt resistor values.  Also the scaling is correct.  I tested with 10k resistor and all calculations seemed more or less accurate.

I think she is ready to go!

smlng commented:
@MrKevinWeiss I cannot merge: no rights to do so, please do on you own

MrKevinWeiss commented:
I increased the version to 0.1.0 when releasing on pip...   I think we will need to do another cleanup before the summit.  Functionality is there though.
